lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,188 @@
+  // Testing on NVDA shows a need to set this as the aria-label in addition to the tooltip
+  const buttonLabel = 'Dismiss Upload';
+  const selectedFileRef = useRef(null);
+
+  function onDismiss() {
+    onDismissProp(file, selectedFileRef.current);
+  }
+
+  return (
+    <span className="nx-selected-file" ref={selectedFileRef}>
+      <span className="nx-selected-file__info">
+        <NxOverflowTooltip>
+          <span className="nx-selected-file__name">{file.name}</span>
+        </NxOverflowTooltip>
+        <span className="nx-selected-file__size">{formatSize(file.size)}</span>
+      </span>
+      <NxTooltip title={buttonLabel}>
+        <button type="button" aria-label={buttonLabel} className="nx-selected-file__dismiss-btn" onClick={onDismiss}>
+          <NxFontAwesomeIcon icon={faTimesCircle} />
+        </button>
+      </NxTooltip>
+    </span>
+  );
+}
+
+const NxFileUpload = forwardRef<HTMLDivElement, Props>(function NxFileUpload(props, ref) {
+  const {
+        className: classNameProp,
+        onChange: onChangeProp,
+        files,
+        isRequired,
+        isPristine,
+        id,
+        disabled,
+        ...attrs
+      } = props,
+      isFileSelected = !!files?.length,
+      { showValidationErrors: formShowValidationErrors } = useContext(FormAriaContext),
+      showValidationErrors = formShowValidationErrors || !isPristine,
+      showError = isRequired && showValidationErrors && !isFileSelected,
+      className = classnames('nx-multi-file-upload', classNameProp),
+      noFileMessageClassName = classnames('nx-multi-file-upload__no-file-message', {
+        'nx-multi-file-upload__no-file-message--invalid': showError
+      }),
+      inputRef = useRef<HTMLInputElement>(null),
+      inputId = useUniqueId('nx-multi-file-upload-input', id),
+      selectedFilesContainerRef = useRef<HTMLDivElement>(null),
+      validationErrorId = useUniqueId('nx-multi-file-upload-validation-error'),
+      [previousFiles, setPreviousFiles] = useState<FileList | null>(files);
+
+  function onChange(evt: FormEvent<HTMLInputElement>) {
+
+    const dataTransferObject = new DataTransfer();
+
+    // get the previous files uploaded
+    if (files) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(files));
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    if (inputRef.current) {
+      inputRef.current.files = dataTransferObject.files;
+      onChangeProp(inputRef.current.files);
+    }
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj : File, selectedFile : HTMLElement | null) {
+    // if the selected file is either the last file in the container or the last remaining file, set focus accordingly
```

Comment:
> accordingly

What are the focus rules that are being implemented here? They should be decided by designers and documented in kaleido as well.

Reply:
> What are the focus rules that are being implemented here? They should be decided by designers and documented in kaleido as well.

I discussed this with design last week. It was decided that when the last file is removed, the focus shifts to either the previous file's dismiss button (if there is one), or the input (if it was the only selected file).  I added in more of an explanation here:  [7376d32](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/7376d32053e47416c0c5a432e8b2482e5c1dda52)